tsdb/db_test.go

```diff
@@ -870,9 +882,17 @@ func TestWALSegmentSizeOptions(t *testing.T) {
 		},
 		// Wal disabled.
 		-1: func(dbDir string, segmentSize int) {
-			if _, err := os.Stat(filepath.Join(dbDir, "wal")); !os.IsNotExist(err) {
-				t.Fatal("wal directory is present when the wal is disabled")
-			}
+			// Check that WAL dir is there.
```

Comment:
WAL cannot be disabled by a user I believe (except the Read only DB). It is only in tests that WAL is disabled at places. And based on how code would be designed for simplicity, it would be easier to always have mmaped chunk rather than it being optional.